polar-core/src/partial/isa_constraint_check.rs

```diff
@@ -65,75 +74,98 @@ impl IsaConstraintCheck {
         // Not comparable b/c one of the matches statements has a LHS that isn't a variable or dot
         // op.
         if constraint_path.is_empty() || proposed_path.is_empty() {
-            return (None, None);
+            return Check::None;
         }
 
-        // a.b.c vs. d
-        if constraint_path
+        let just_vars = constraint_path.len() == 1
+            && proposed_path.len() == 1
+            && matches!(&constraint.args[0].value().as_symbol(), Ok(Symbol(_)))
+            && matches!(&self.proposed.args[0].value().as_symbol(), Ok(Symbol(_)));
+
+        // FIXME(gw): this logic is hard to follow!
```

Comment:
Can you explain what this logic is doing?

Reply:
first exclude a bunch of cases where we don't care about this constraint: it's not an `Isa`, it's not checking variables, it's checking two unrelated variables, or the paths don't match. then check for the cases we do care about: the sides are the same (is subclass check), or we already have a constraint on a prefix of an access path (isa with path check).

the only real difference from before is (supposed to be) that now we check if the "root" variable of an existing constraint is an alias for the root of the proposed constraint, instead of if it's identical.

Reply:
lol gabe